### PR TITLE
Swope 13 WIP-do not merge

### PR DESCRIPTION
**Pull request by swopebe**

### CODE REVIEW

File: health_metrics/accessing_metrics.adoc [hunk 1]
```diff
@@ -0,0 +1,126 @@
+[#accessing-metrics]
+= Accessing the metrics service
+
+You can use metrics to monitor the health of your fleet across components. See the following documentation:
```
Comment (swopebe) [inline health_metrics/accessing_metrics.adoc]: We need a good description here, please commit.

File: health_metrics/accessing_metrics.adoc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+controller_runtime_max_concurrent_reconciles{controller="klusterlet-token-controller"} 1
+controller_runtime_max_concurrent_reconciles{controller="subscription-controller"} 1
+controller_runtime_reconcile_errors_total{controller="helmrelease-controller"} 0
+controller_runtime_reconcile_errors_total{controller="klusterlet-token-controller"} 22
+controller_runtime_reconcile_errors_total{controller="subscription-controller"} 0
+----
+
+[#accessing-standalone-metrics]
+== Accessing the standalone metrics service
+//need descriptions after header
```
Comment (chenz4027) [inline health_metrics/accessing_metrics.adoc]: To view the collected metrics, you would need to follow the following steps to expose the service on the hub cluster.Note: standalone application subscription will only deploy to the hub cluster. As it won't deploy to managed clusters, there is no need to create manifest work, thus no propagation time.

File: health_metrics/accessing_metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,126 @@
+[#accessing-metrics]
+= Accessing the metrics service
+
+You can use metrics to monitor the health of your fleet across components. See the following documentation:
+//just an outline, these files do not exist until feedback from the team about what to include from open TOC
+
+[#accessing-hub-metrics]
+== Accessing the hub cluster metrics service
+//need descriptions after header
```
Comment (chenz4027) [inline health_metrics/accessing_metrics.adoc]: To view the collected metrics, you would need to follow the following steps to expose the service on the hub cluster.

File: health_metrics/accessing_metrics.adoc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+workqueue_adds_total{name="addon-hook-deploy-controller"} 9
+workqueue_adds_total{name="addon-install-controller"} 12
+workqueue_adds_total{name="addon-registration-controller"} 10
+workqueue_adds_total{name="cluster-management-addon-controller"} 9
+workqueue_adds_total{name="mcmhub-subscription-controller"} 0
+----
+
+[#accessing-managed-metrics]
+== Accessing the managed cluster metrics service
+//need descriptions after header
```
Comment (chenz4027) [inline health_metrics/accessing_metrics.adoc]: To view the collected metrics, you would need to follow the following steps to expose the service on the managed cluster.

File: troubleshooting/trouble_metric_no_hub.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
+[#troubleshooting-hub-cluster-metrics-missing]
+= Troubleshooting hub cluster metrics missing
+//Need description info
```
Comment (chenz4027) [inline troubleshooting/trouble_metric_no_hub.adoc]: Content from https://github.com/open-cluster-management-io/multicloud-operators-subscription/blob/main/docs/metrics.md#hub-cluster-metrics-service-missing should be enough.

**Review by chenz4027 — COMMENTED** [review]
Added a few descriptions. @xiangjingli @TomerFi to see if anything is missing.

File: health_metrics/accessing_metrics.adoc (hunk 1 above)
Comment (bjoydeep) [reply health_metrics/accessing_metrics.adoc]: You can use metrics to monitor `the health of ACM components`. That probably would be better. The `Health of your fleet` is done by ACM Observability itself. This is more internal facing

Comment (bjoydeep) [reply health_metrics/accessing_metrics.adoc:4]: Also this is only for Application LC components, right? There is also health metrics from GRC, Server Foundation, Observability which are exposed differently. Do we want to bring them altogether here - or just call it Application LC health metrics?

Comment (swopebe) [reply health_metrics/accessing_metrics.adoc]: We discussed this in the first call, see that I set up a generic guide now for everything.

File: health_metrics/accessing_metrics.adoc (hunk 3 above)
Comment (dockerymick) [reply health_metrics/accessing_metrics.adoc:9]: Added

File: health_metrics/custom_metrics.adoc [hunk 6]
```diff
@@ -0,0 +1,76 @@
+[#custom-metrics]
+= Custom metrics
+//need descriptions after header
```
Comment (dockerymick) [inline health_metrics/custom_metrics.adoc]:
```suggestion

You can manage your custom metrics for your hub clusters and managed clusters. Continue reading to understand which metrics can be scrapped from both hub and managed clusters, and where to manage custom metrics.
```

File: health_metrics/custom_metrics.adoc [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+[#custom-metrics-hub]
+== Hub cluster custom metrics
+//need descriptions after header
```
Comment (dockerymick) [inline health_metrics/custom_metrics.adoc]:
```suggestion

You can expose specific hub cluster metrics to be scrapped. View the following table for descriptions of the metrics that can be scrapped from the hub cluster:
```

File: health_metrics/custom_metrics.adoc [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+The following metrics can be scrapped from _Hub Cluster_:
```
Comment (dockerymick) [inline health_metrics/custom_metrics.adoc]:
```suggestion

```

File: health_metrics/custom_metrics.adoc [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+latency 
+|_subscription_namespace__subscription_name_
+|propagation_failed_time 
+|Histogram of failed propagation latency
+|_subscription_namespace__subscription_name_
+|===
+
+[#custom-metrics-managed]
+== Managed cluster custom metrics
+//need descriptions after header
```
Comment (dockerymick) [inline health_metrics/custom_metrics.adoc]:
```suggestion

Similar to the hub cluster metrics, view the following table for descriptions of the metrics that can be scrapped from your managed clusters:
```

File: health_metrics/custom_metrics.adoc [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+
+The following metrics can be scrapped from _Managed Clusters_:
```
Comment (dockerymick) [inline health_metrics/custom_metrics.adoc]:
```suggestion

```

File: health_metrics/custom_metrics.adoc [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+|_subscription_namespace__subscription_name_
+|local_deployment_failed_time 
+|Histogram of failed local deployment
+latency 
+|_subscription_namespace__subscription_name_
+|===
+
+[#custom-metrics-observability-]
+== Observability custom metrics
+//need descriptions after header
```
Comment (dockerymick) [inline health_metrics/custom_metrics.adoc]:
```suggestion
Observability custom metrics are listed in the `metrics_list.yaml` section of the `MultiClusterObservability` resource. Before you add a custom metric, verify that `mco observability` is enabled. Read _Adding custom metrics_ for more information. 
```

Currently, the section **Adding custom metrics** is [here](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/observability/customize_observability.adoc#adding-custom-metrics). I am still working on modularizing the observability doc. I can imagine _Adding custom metrics_ as a file within the `health_metrics` folder. @swopebe @dislbenn let me know your thoughts

File: troubleshooting/trouble_metric_no_managed.adoc [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
++
+[source,bash]
+----
+oc get svc -n open-cluster-management hub-subscription-metrics
+----
+
+The metrics service is now available on your managed cluster.
+
+//is there a reason why we might want the user to be aware of the Port difference?
+//Note the use of the non-conventional port 8388 (instead of the conventional 8381).
```
Comment (dockerymick) [inline troubleshooting/trouble_metric_no_managed.adoc]: @chenz4027 @dislbenn

File: health_metrics/custom_metrics.adoc [hunk 13]
```diff
@@ -0,0 +1,77 @@
+[#custom-metrics]
+= Custom metrics
```
Comment (swopebe) [inline health_metrics/custom_metrics.adoc]: This to be moved to solostron/metrics-chronicle

File: health_metrics/main.adoc [hunk 14]
```diff
@@ -0,0 +1,8 @@
+include::modules/common-attributes.adoc[]
+
+= Health metrics
+
+include::health_intro.adoc[leveloffset=+1]
+include::accessing_metrics.adoc[leveloffset=+2]
+include::scrapping.adoc[leveloffset=+2]
+
```
Comment (dockerymick) [inline health_metrics/main.adoc:7]: @dislbenn @bjoydeep @swopebe I have the following topics in the Customizing observability doc:

* [Adding custom metrics](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/observability/customize_observability.adoc#adding-custom-metrics)
    * [Adding user workload metrics](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/observability/customize_observability.adoc#adding-user-workload-metrics)
    * [Exporting metrics to external endpoints](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/observability/customize_observability.adoc#exporting-metrics-to-external-endpoints)
    * [Removing default metrics](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/observability/customize_observability.adoc#removing-default-metrics)
* [Viewing the status of metric export](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/observability/customize_observability.adoc#viewing-the-status-of-metrics-export)

I think we should consider renaming `accessing_metrics.adoc` to `access_metric_service.adoc`. Then add a new file named `viewing_metrics.adoc`, where we describe how users can access specific metrics and data. 

Let me know yall thoughts. If you agree, I volunteer to move the info into the new book

File: troubleshooting/trouble_metric_no_managed.adoc (hunk 12 above)
Comment (dockerymick) [reply troubleshooting/trouble_metric_no_managed.adoc]: Target port doesn't need to be documented here. Confirmed by @dislbenn

File: health_metrics/health_metrics.adoc [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: prometheus-k8s-monitoring
+subjects:
+- kind: ServiceAccount
+  name: prometheus-k8s
+  namespace: monitoring
+----
+
+. Verification In `+Prometheus+`’s dashboard, run the following query to look up
```
Comment (dockerymick) [inline health_metrics/health_metrics.adoc]:
```suggestion
. Verify which metrics are reported by the Subscription Operator Metrics Service from the _Prometheus_ dashboard. Run the following command:
```

File: health_metrics/health_metrics.adoc [hunk 16]
```diff
@@ -0,0 +1,339 @@
...
+metrics reported by the Subscription Operator Metrics Service:
```
Comment (dockerymick) [inline health_metrics/health_metrics.adoc]:
```suggestion

```

**Review by dockerymick — APPROVED** [review]
I'm stuck on the Linter too. I made a couple of updates to help with aligning the steps. Otherwise this looks good to me! Thanks so much !

**Comment by swopebe** [comment]
Thanks @dockerymick -- It's eating my time and doesn't seem to make sense so I am gonna merge and see what Pantheon gets me.

**Pull request merged**